packages/appkit/src/client/appkit-base-client.ts

```diff
@@ -155,6 +156,31 @@ export abstract class AppKitBaseClient {
     PublicStateController.set({ initialized: true })
 
     await this.syncExistingConnection()
+    // Check allowed origins only if email or social features are enabled
+    if (
+      OptionsController.state.features?.email ||
+      (Array.isArray(OptionsController.state.features?.socials) &&
+        OptionsController.state.features?.socials.length > 0)
+    ) {
+      await this.checkAllowedOrigins()
+    }
+  }
+
+  private async checkAllowedOrigins() {
+    const allowedOrigins = await ApiController.fetchAllowedOrigins()
+    if (allowedOrigins && typeof window !== 'undefined') {
```

Comment:
Can we use `CoreHelperUtil.isClient()` instead?

```suggestion
    if (allowedOrigins && CoreHelperUtil.isClient()) {
```